### PR TITLE
Enable all possible attr for report string control

### PR DESCRIPTION
**Pull request by janezj**

### CODE REVIEW

**Comment by awkay** [comment]
See com.fulcrologic.rad.rendering.semantic-ui.controls.control for the standard wrapper. You're right that the control in question isn't doing it, but there is an established key for this: `:user-props`. That wrapper is used for a lot of the controls so it gets done automatically. The one you're looking at was a one-off and I forgot to do it. Could you change it to that established pattern?

**Comment by awkay** [comment]
I should also get around to putting documented keys in `control-options` so that it is easier to find these. When I originally did these the verdict was out on where the boundary is on control options and how wild the list of options might get. But the core standard ones would be good to have in a ns.

**Comment by janezj** [comment]
Done.

I noticed that user-props is sometimes the first argument of (merge
user-props ...), sometimes the last arg (boolean, enumerated)

File: src/main/com/fulcrologic/rad/semantic_ui_options.cljc [hunk 1]
```diff
@@ -60,6 +60,17 @@
...
 
+(def report-string-control-attributes
+  "Provide dom/input attributes for report string  control
+   Example:
+     ::max       {:type          :string
+                  :local?        true
+                  :label         \"Max\"
+                  :default-value \"10\"
+                  suo/report-string-control-attributes
+                  {:type \"number\" :min \"4\" :step \"2\" :max \"20\"} }"
```
Comment (awkay) [inline src/main/com/fulcrologic/rad/semantic_ui_options.cljc:71]: This is no longer used....

**Comment by awkay** [comment]
> Done. I noticed that user-props is sometimes the first argument of (merge user-props ...), sometimes the last arg (boolean, enumerated)


Thanks. Yeah, on some controls I guess I was thinking "them overriding those things is silly", whereas on others I guess I assumed it might be desirable. Handlers, for example, probably cannot be usefully overridden untill we allow user-props to be a function (which we could do, but that isn't how it is currently).

Probably the pattern should be:

```
(let [local-props {:onBlur ...}]
   (merge local-props (?! user-props this local-props))
  ...
```

so that user-props could be used to completely take over some part of the functionality, including being able to optionally wrap the control-generated items (e.g. you'd be able to get the plugin-defined onBlur and wrap it). That can be a future task :)

File: src/main/com/fulcrologic/rad/rendering/semantic_ui/components.cljc [hunk 2]
```diff
@@ -42,10 +42,10 @@
 (defsc WrappedDropdown [this {:keys [onChange value multiple] :as props}]
   {:initLocalState (fn [this]
                      #?(:cljs
-                        (let [xform-options (memoize (fn [options]
```
Comment (awkay) [inline src/main/com/fulcrologic/rad/rendering/semantic_ui/components.cljc:45]: Why did you remove memoize? The transit conversion can be slow on large numbers of options. We should at least be using enc/memoize with a timeout, as this was added to address a real performance issue with real apps.

File: src/main/com/fulcrologic/rad/rendering/semantic_ui/report.cljc [hunk 3]
```diff
@@ -257,12 +257,12 @@
                                (dom/a {:onClick (fn [evt]
                                                   (evt/stop-propagation! evt)
                                                   (report/sort-rows! report-instance column))}
-                                 (str label)
```
Comment (awkay) [inline src/main/com/fulcrologic/rad/rendering/semantic_ui/report.cljc:260]: Oh boy...technically a breaking change. If someone used a keyword, say, for a label, then their code would have worked before and will crash now. I think I'm going to let this slide because having DOM elements as a label is more important in this case, and it was a genuine oversight. BUT, needs to be listed in the commit message as a BREAKING CHANGE.

**Review by awkay — APPROVED** [review]

**Comment by awkay** [comment]
Merged/fixed manually
